### PR TITLE
Miner holds it's own copy of spec/engine

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

File: sync/src/tests/helpers.rs [hunk 1]
```diff
@@ -93,7 +93,7 @@ impl TestNet {
 		for _ in 0..n {
 			net.peers.push(TestPeer {
 				chain: TestBlockChainClient::new(),
-				sync: ChainSync::new(SyncConfig::default(), Miner::new(false)),
+				sync: ChainSync::new(SyncConfig::default(), Arc::new(Miner::default())),
```
Comment (gavofyork) [inline sync/src/tests/helpers.rs]: prefer to keep this explicit (i.e. `Miner::new` with a well-defined spec param)

**Comment by gavofyork** [comment]
has this been tested on a mining rig?

**Comment by NikVolf** [comment]
Nope, 
Care to do a quick test, @General-Beck ?

**Comment by General-Beck** [comment]
I'll try tomorrow

**Comment by gavofyork** [comment]
only needs to be tested on morden. doesn't take long to mine a block on there.

**Comment by General-Beck** [comment]
test passed in mainnet

**Pull request merged**